### PR TITLE
Issue/fix compilation warnings

### PR DESCRIPTION
**Pull request by mateuszkulak**

### CODE REVIEW

File: src/Core/LeanCode.Components.Startup/IHostBuilderExtensions.cs [hunk 1]
```diff
@@ -110,7 +111,7 @@ public static class IHostBuilderExtensions
 
                 if (context.HostingEnvironment.IsDevelopment())
                 {
-                    loggerConfiguration.WriteTo.Console();
+                    loggerConfiguration.WriteTo.Console(formatProvider: CultureInfo.InvariantCulture);
```
Comment (jakubfijalkowski) [inline src/Core/LeanCode.Components.Startup/IHostBuilderExtensions.cs:114]: Can we remove the name of the parameter? For a single parameter it does not add to the meaning.

File: src/Tools/LeanCode.CodeAnalysis/CodeFixProviders/AddAuthorizationAttributeCodeFixProvider.cs [hunk 2]
```diff
@@ -49,6 +49,10 @@ Compilation compilation
     )
     {
         var baseAttribute = compilation.GetTypeByMetadataName(AuthorizeWhenAttribute);
+
+        if (baseAttribute is null)
+            return Enumerable.Empty<(string Type, string Namespace)>();
```
Comment (jakubfijalkowski) [inline src/Tools/LeanCode.CodeAnalysis/CodeFixProviders/AddAuthorizationAttributeCodeFixProvider.cs]:
```suggestion
        if (baseAttribute is null)
        {
            return Enumerable.Empty<(string Type, string Namespace)>();
        }
```

File: src/Tools/LeanCode.StyleCop/LeanCode.StyleCop.csproj [hunk 3]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="StyleCop.Analyzers" PrivateAssets="None" />
+    <PackageReference Include="StyleCop.Analyzers" PrivateAssets="None"  Version="1.1.118"/>
```
Comment (jakubfijalkowski) [inline src/Tools/LeanCode.StyleCop/LeanCode.StyleCop.csproj]:
```suggestion
    <PackageReference Include="StyleCop.Analyzers" PrivateAssets="None"  Version="1.2.0-beta.435" />
```

StyleCop needs to be on the newest version to work - the stable one is 4 years old.

File: test/LeanCode.IntegrationTests/App/ApiModule.cs [hunk 4]
```diff
@@ -48,7 +48,7 @@ public override void ConfigureServices(IServiceCollection services)
             })
             .AddJwtBearer(cfg =>
             {
-                cfg.Authority = UrlHelper.Concat(config.GetValue<string>("InternalBase"), "auth");
+                cfg.Authority = UrlHelper.Concat(config.GetValue<string>("InternalBase") ?? "", "auth");
```
Comment (jakubfijalkowski) [inline test/LeanCode.IntegrationTests/App/ApiModule.cs]:
```suggestion
                cfg.Authority = UrlHelper.Concat(config.GetValue<string>("InternalBase")!, "auth");
```

It should always be there.

File: test/Testing/LeanCode.IntegrationTestHelpers.Tests/IntegrationTestsConfigurationTests.cs [hunk 5]
```diff
@@ -20,8 +20,8 @@ public void Custom_configuration_keys_are_respected()
 
         var config = new ConfigurationBuilder().Add(configurationOverrides).Build();
 
-        Assert.NotEmpty(config.GetValue<string>(customConnectionStringKey));
-        Assert.NotEmpty(config.GetValue<string>(customInternalBaseKey));
-        Assert.NotEmpty(config.GetValue<string>(customPublicBaseKey));
+        Assert.NotEmpty(config.GetValue<string>(customConnectionStringKey) ?? string.Empty);
+        Assert.NotEmpty(config.GetValue<string>(customInternalBaseKey) ?? string.Empty);
+        Assert.NotEmpty(config.GetValue<string>(customPublicBaseKey) ?? string.Empty);
```
Comment (jakubfijalkowski) [inline test/Testing/LeanCode.IntegrationTestHelpers.Tests/IntegrationTestsConfigurationTests.cs]:
```suggestion
        Assert.NotEmpty(config.GetValue<string>(customConnectionStringKey)!);
        Assert.NotEmpty(config.GetValue<string>(customInternalBaseKey)!);
        Assert.NotEmpty(config.GetValue<string>(customPublicBaseKey)!);
```

Same here - these are tests, they make _some_ assumptions.

**Review by jakubfijalkowski — CHANGES_REQUESTED** [review]

File: src/Core/LeanCode.Components.Startup/IHostBuilderExtensions.cs (hunk 1 above)
Comment (mateuszkulak) [reply src/Core/LeanCode.Components.Startup/IHostBuilderExtensions.cs:114]: I don't think we can. This function takes several parameters so it tries to convert formatProvider to LogEventLevel if we don't specify which parameter it is.

Comment (jakubfijalkowski) [reply src/Core/LeanCode.Components.Startup/IHostBuilderExtensions.cs:114]: Ahhh, got it!

**Review by ZoltePudeleczko — APPROVED** [review]

**Review by jakubfijalkowski — APPROVED** [review]

**Pull request merged**